src/components/groups/group.middlewares.ts

```diff
@@ -120,6 +120,27 @@ export const createICSEvent = (req: Request, res: Response) => {
   })
 }
 
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function isValidHttpsUrl(str) {
+  let url
+  try {
+    url = new URL(str)
+  } catch (_) {
+    return false
+  } // not catching bad top lvl domain (1 character)
+
+  const pattern = new RegExp('^(https?:\\/\\/)?'+ // protocol
+    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
+    '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
+    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ // port and path
+    '(\\?[;&a-z\\d%_.~+=-]*)?'+ // query string
+    '(\\#[-a-z\\d_]*)?$','i') // fragment locator
+  // not allowing '(' and ')'
+  // catching 1 character TLD
+
+  return !!pattern.test(str) && url.protocol === 'https:'
```

Comment:
I think `pattern.test(str)` returns a boolean always (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) so the double exclamation mark is not needed for casting.

Reply:
Yep, those aren't needed